Add TGLF SAT123 and saturation parameter calculations

tomneiser opened the pull request:
This adds the ability to calculate saturated fluxes from quasilinear weight, growth rate and ky spectra, and relevant plasma parameters in `input.tglf`. The saturation rules were developed by Gary Staebler _et al_ for SAT12, and @hgd511 and @fcasson _et al_ for SAT3. This code is very fortran-y, which is deliberate since it is easier to update in response to changes to the saturation rules. I've added a test `test_tglf.py` that can also serve as a minimal working example. Over time the test can be modified to use pyro objects (I wasn't able to load the TGLF template output using pyro at the time of this writing).

tomneiser committed:
tglf satparams functions for dev from omfit

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
SAT3 model

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
Diagnostics:TGLF python update kymax search

tomneiser committed:
Diagnostics:TGLF python update kymax search

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
Diagnostics:minor edit

tomneiser committed:
Merge branch 'unstable' into feature/tglf_satparams

tomneiser committed:
flake8 fixes

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
Added flux integral and tests

tomneiser committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/tglf_satparams' into feature/tglf_satparams

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
flake8 formatting

tomneiser committed:
more flake8

tomneiser committed:
last flake8 test

tomneiser committed:
still last flake8 test

bpatel2107 commented:
This looks like a good start! I think main issue is that this only works using TGLF inputs, whilst we would want this to work with all inputs. Ideally we'd like to use existing `pyro` object to define the inputs to the saturation rule as you say so I think it makes sense to merge into unstable when that is achieved.

For example we already calculate things like local geometry parameter like `Q_LOC` and `Q_PRIME_LOC` and even derivative of flux surface values like   `$\partial Z / \partial r$` etc. in `local_geometry`. `metric_terms` also calculates things like `grad_r` too. Essentially everything in `get_sat_params` should already exist or be trivial to obtain.

I think we would want something like `intensity_sat` to read in a `PyroScan` object which has been done over `ky` or something equivalent. Then we could take a GS2/CGYRO/GENE scan and calculate the sat rules from it.  May be worth arranging a meeting to discuss how to get everything out of pyro from this

tomneiser committed:
Merge branch 'unstable' into feature/tglf_satparams

tomneiser committed:
Merge branch 'unstable' into feature/tglf_satparams

tomneiser committed:
compatibility with use_ave_ion_grid

tomneiser committed:
bugfix for SAT3py

tomneiser committed:
Merge branch 'unstable' into feature/tglf_satparams

tomneiser committed:
tglf:satpy:constrain units and rho_ion

tomneiser committed:
fix typo in use_ave_ion_grid

tomneiser committed:
[skip ci] Apply black changes

bpatel2107 committed:
Fix CGYRO writing when including adiabatic species

physycola committed:
fix handling of adiabatic species when writing input file

physycola committed:
Merge branch 'enhancements/gx_v2' of https://github.com/pyro-kinetics/pyrokinetics into enhancements/gx_v2

bpatel2107 committed:
Update README.md

bpatel2107 committed:
Merge pull request #416 from pyro-kinetics/enhancements/gx_v2

Addition of GX

bpatel2107 committed:
Tidy up flux normalisation

bpatel2107 committed:
Merge branch 'unstable' into bugfix/GKOutput_flux_field

bpatel2107 committed:
Avoid double normalising fluxes/moments

bpatel2107 committed:
Flake8 fix

bpatel2107 committed:
CGYRO update QL flux when nky > 1

bpatel2107 committed:
CGYRO update QL flux when nky > 1

bpatel2107 committed:
Add phase to amplitude calc

bpatel2107 committed:
Fix sign convention CGYRO

bpatel2107 committed:
Fix sign on CGYRO flux

bpatel2107 committed:
Renormalise linear weights at each time

bpatel2107 committed:
Updates for tests

bpatel2107 committed:
Update golden answer and small tweaks

bpatel2107 committed:
Fix theta_star phase calculator for any size

bpatel2107 committed:
Allow for case without theta0 in MG sat and add legacy flux defn in GS2

bpatel2107 committed:
Merge pull request #418 from pyro-kinetics/bugfix/mg_sat_no_theta0

Allow for case without theta0 in MG sat and add legacy flux defn in GS2

bpatel2107 committed:
Update GX/CGYRO flux_norm, add old stella flux

bpatel2107 committed:
Pass on output_convention to GKOutput in STELLA

bpatel2107 committed:
[skip ci] Apply black changes

bpatel2107 committed:
Promote theta to float64 and update trapz rule in GS2

bpatel2107 committed:
black fix gs2.py

bpatel2107 committed:
Add gradr correction

tomneiser committed:
Merge remote-tracking branch 'origin/unstable' into feature/tglf_satparams

tomneiser committed:
Merge remote-tracking branch 'origin/unstable' into feature/tglf_satparams

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
fix get_zonal and update sat3

tomneiser committed:
fix get_zonal and update sat3

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
test tglf saturation

tomneiser committed:
test tglf saturation

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
[skip ci] Apply black changes

bpatel2107 committed:
Fix normalisation of fluxes and pyroscan for GX

bpatel2107 committed:
Update GX tests

d7919 committed:
Update gs2.py to force fapar and fbpar to zero if beta is zero

This replicates logic inside GS2 which disables the apar and bpar flag if beta is zero

d7919 committed:
[skip ci] Apply black changes

d7919 committed:
Update gs2.py to give s_hat_input a default of shat if present

Duplicates logic in GS2 about the default for s_hat_input.

d7919 committed:
[skip ci] Apply black changes

d7919 committed:
Update input.gs2 to remove diagnostics_config namelist

This namelist hasn't been used in many years

d7919 committed:
Update gs2.in to remove diagnostics_config and merge parameters to knobs

The diagnostics_config hasn't been used in many years and the values in the parameters namelist should appear in knobs now instead

bpatel2107 committed:
Merge branch 'unstable' into bugfix/GKOutput_flux_field

Conflicts:
	src/pyrokinetics/gk_code/gs2.py

bpatel2107 committed:
Fix merge gs2 flux

bpatel2107 committed:
Switch logic in pyroscan to read TGLF eigenvalue solver

bpatel2107 committed:
Fix sign issue in linear TGLF and force electrons to be written first

bpatel2107 committed:
Handle negative BTCCW better

bpatel2107 committed:
Merge branch 'bugfix/GKOutput_flux_field' of github.com:pyro-kinetics/pyrokinetics into bugfix/GKOutput_flux_field

bpatel2107 committed:
Actually fix CGYRO sign conventions

bpatel2107 committed:
Update tests

bpatel2107 committed:
Add px0 to moments

bpatel2107 committed:
Apply conjugate to CGYRO with NL run too

bpatel2107 committed:
Merge pull request #421 from pyro-kinetics/d7919-patch-1

Update gs2.py to force fapar and fbpar to zero if beta is zero

bpatel2107 committed:
Merge pull request #422 from pyro-kinetics/d7919-patch-2

Update gs2.py to give s_hat_input a default of shat if present

bpatel2107 committed:
Merge pull request #424 from pyro-kinetics/d7919-patch-3

Update input.gs2 to remove diagnostics_config namelist

bpatel2107 committed:
Merge pull request #423 from pyro-kinetics/d7919-patch-4

Update gs2.in to remove diagnostics_config and merge parameters to knobs

bpatel2107 committed:
Merge pull request #364 from pyro-kinetics/bugfix/GKOutput_flux_field

Direct matching of linear GKOutput between codes

bpatel2107 committed:
Dont remove all moments in cgyro mxh

bpatel2107 committed:
Try switch to MXH in tglf

bpatel2107 committed:
Quick hack for negative q CGYRO

bpatel2107 committed:
Fix GENE Bpol defn when using tracer_efit

bpatel2107 committed:
Ensure Rmaj/Z0/rho are consistent in MXH and make theta go from 0 to 2pi

bpatel2107 committed:
Get correct norm_convention for GENE outputs (like other codes)

bpatel2107 committed:
Use physical units to define rhoref_custom_name rather than simulation units

bpatel2107 committed:
Merge branch 'unstable' into bugfix/gene_mxh_map

# Conflicts:
#	src/pyrokinetics/gk_code/gene.py

bpatel2107 committed:
Find Z0_ind using outboard points only

physycola committed:
force sign of ky to be positive in cgyro.py

bpatel2107 committed:
Fix theta defn from get_coords and reshape given fields

physycola committed:
[skip ci] Apply black changes

physycola committed:
merged changes into previous line

physycola committed:
[skip ci] Apply black changes

bpatel2107 committed:
Add option to read legacy STELLA inputs

physycola committed:
blank commit for tests

bpatel2107 committed:
Merge pull request #427 from pyro-kinetics/bugfix/force_CGYRO_positive_ky

Force the sign of ky to be positive when loading gk_output

bpatel2107 committed:
Raise error if no gene geometry file found

bpatel2107 committed:
Correct default parameters_params

bpatel2107 committed:
Merge pull request #426 from pyro-kinetics/bugfix/gene_mxh_map

Bugfix/gene mxh map

bpatel2107 committed:
Use existing variable

bpatel2107 committed:
[skip ci] Apply black changes

bpatel2107 committed:
Remove unnecessary variable

bpatel2107 committed:
Merge branch 'bugfix/cgyro_field_negative_q' of github.com:pyro-kinetics/pyrokinetics into bugfix/cgyro_field_negative_q

bpatel2107 committed:
:Merge branch 'unstable' into bug-fix

bpatel2107 committed:
Better logic for multiple gradients and handle 0s

bpatel2107 committed:
[skip ci] Apply black changes

bpatel2107 committed:
Merge pull request #392 from pyro-kinetics/bug-fix

changed np.isclose() to np.allclose() for calculating normed gradients

bpatel2107 committed:
Handle adiabatic electrons, circ geometry, no collisions and nonlinear box

bpatel2107 committed:
Add nml variable needed for test_normalisation.py

bpatel2107 committed:
GENE set beta_prime correctly if amhd=-1

bpatel2107 committed:
Fix MXH setting and add MXH TGLF into roundtrip test

bpatel2107 committed:
Update test to account for correct sign in TGLF linear

physycola committed:
fixed line for constructing a psi_n grid for species charge

physycola committed:
Merge pull request #435 from pyro-kinetics/bugfix/get_charge_fix

get_charge() method fix

bpatel2107 committed:
Merge branch 'unstable' into bugfix/tglf_pyroscan

Conflicts:
	tests/gk_code/golden_answers/tglf_linear_output_560f9b79.netcdf4
	tests/gk_code/test_gk_output_reader_tglf.py

bpatel2107 committed:
Merge pull request #434 from pyro-kinetics/bugfix/tglf_pyroscan

Bugfix/tglf pyroscan

bpatel2107 committed:
Merge pull request #428 from pyro-kinetics/enhancements/stella_backward_compat

Stella namelist backward compatibility

bpatel2107 committed:
Merge pull request #433 from pyro-kinetics/bugfix/gene_amhd

GENE set beta_prime correctly if amhd=-1

bpatel2107 committed:
Add in flux surface area/vol calc

bpatel2107 committed:
Simpler defn for vol

bpatel2107 committed:
Simplify surf area calc

bpatel2107 committed:
Update docstring

bpatel2107 committed:
Add derivatives and a test for it

bpatel2107 committed:
Merge pull request #436 from pyro-kinetics/enhancements/flux_surface_volumes

Enhancements/flux surface volumes

physycola committed:
omprimfac was being improperly set; should be default of 1.0 irrespective of the sign of the rotation, given that it is a muliplier that gets applied after the PVG term is calculated.

physycola committed:
Revert "omprimfac was being improperly set; should be default of 1.0 irrespective of the sign of the rotation, given that it is a muliplier that gets applied after the PVG term is calculated."

This reverts commit d06a17a1caaeaf1f51a19df1dc6ef259109b4e13.

physycola committed:
Fixed double minus-sign error spotted by Bhavin.

physycola committed:
[skip ci] Apply black changes

physycola committed:
Blank commit for automatic tests

physycola committed:
Merge branch 'bugfix/gs2_omprimfac' of https://github.com/pyro-kinetics/pyrokinetics into bugfix/gs2_omprimfac

bpatel2107 committed:
Merge pull request #438 from pyro-kinetics/bugfix/gs2_omprimfac

Fix to setting of omprimfac in gs2.py

bpatel2107 committed:
Add in splines for psi_tor and rho_tor

physycola committed:
Bugfix to prevent AttributeError

physycola committed:
Added lithium to ion_species_selector

physycola committed:
Removed erroneous print statement

physycola committed:
Added the "DT_5050" for a 50/50 mixture of deuterium/tritium (Z = 1.0, A = 2.5)

physycola committed:
Merge pull request #440 from pyro-kinetics/bugfix/pfile_loading

Bugfix to prevent AttributeError

physycola committed:
Possibly erroneous calculation of ntheta in gs2_geometry_terms

bpatel2107 committed:
Add check for LCFS a minor in case of divertor legs

bpatel2107 committed:
Switch to np.isclose

bpatel2107 committed:
use shapely to remove divertor legs

bpatel2107 committed:
[skip ci] Apply black changes

bpatel2107 committed:
Add shapely to requirements

bpatel2107 committed:
Merge branch 'bugfix/lcfs_contour_fit' of github.com:pyro-kinetics/pyrokinetics into bugfix/lcfs_contour_fit

bpatel2107 committed:
flake8 fix

bpatel2107 committed:
Append contour as Z,R

bpatel2107 committed:
Fix tests with correct theta grid on get_gs2_geom and add new gs2 output template file

bpatel2107 committed:
black fix

bpatel2107 committed:
Pin pint-xarray for now

bpatel2107 committed:
Stricter requirement on pint-xarray

bpatel2107 committed:
Merge pull request #442 from pyro-kinetics/bugfix/gs2_geometry_terms

Possibly erroneous calculation of ntheta in gs2_geometry_terms

bpatel2107 committed:
simple test for psi_tor

bpatel2107 committed:
Merge pull request #430 from pyro-kinetics/bugfix/cgyro_field_negative_q

CGYRO signs of Bt/Ip/q

bpatel2107 committed:
Merge branch 'unstable' into enhancements/equilibrium_tor_flux

bpatel2107 committed:
Merge pull request #439 from pyro-kinetics/enhancements/equilibrium_tor_flux

Add in splines for psi_tor and rho_tor

physycola committed:
Changed the logic in pfile.py to return "hydrogenic" when a potential mix of hydrogenic species. The changes in normalisations.py currently gives the error `pint.errors.UndefinedUnitError: 'mref_hydrogenic_EQDSK_COCOS_050000' is not defined in the unit registry`.

physycola committed:
[skip ci] Apply black changes

bpatel2107 committed:
Merge branch 'unstable' into bugfix/hydrogenic_species

bpatel2107 committed:
Add hydrogenic mref manually and make a test using pfile

bpatel2107 committed:
Merge pull request #447 from pyro-kinetics/bugfix/hydrogenic_species

Bugfix/hydrogenic species

bpatel2107 committed:
Add test for quasineutrality before writing gk_files, modify jetto.py to account for beam, make template pfile quasineutral

bpatel2107 committed:
[skip ci] Apply black changes

bpatel2107 committed:
Add corrected input.gacode (based off of JETTO) with beam ions

bpatel2107 committed:
Merge remote-tracking branch 'origin/feature/enforce_qn' into feature/enforce_qn

bpatel2107 committed:
Merge branch 'unstable' into feature/enforce_qn

bpatel2107 committed:
Update answers for pfile hydrogenic

bpatel2107 committed:
Merge pull request #444 from pyro-kinetics/bugfix/lcfs_contour_fit

Add check for LCFS a minor in case of divertor legs

bpatel2107 committed:
Add units to PyroScan output and add .to method for converting

bpatel2107 committed:
[skip ci] Apply black changes

bpatel2107 committed:
Handle case without units, add warning if no units and make format more consistent with Pyro.GKOutput

bpatel2107 committed:
Merge branch 'enhancements/pyroscan_units' of github.com:pyro-kinetics/pyrokinetics into enhancements/pyroscan_units

bpatel2107 committed:
[skip ci] Apply black changes

bpatel2107 committed:
Move unit check to inside load_gk_output

physycola committed:
Fixed cases where division by zero would occur due to finite precision of pfile data

physycola committed:
[skip ci] Apply black changes

physycola committed:
Blank commit for triggering online tests

bpatel2107 committed:
Add units to pyroscan json

bpatel2107 committed:
Add warning when adding units during load_gk_output

bpatel2107 committed:
Check for units when setting parameter map and convert to correct units at the end

bpatel2107 committed:
Merge pull request #452 from pyro-kinetics/bugfix/pfile_divide_by_zero

Bugfix/pfile divide by zero

bpatel2107 committed:
Fix pyroscan test

bpatel2107 committed:
Merge branch 'unstable' into enhancements/pyroscan_units

bpatel2107 committed:
Merge pull request #450 from pyro-kinetics/enhancements/pyroscan_units

Add units to PyroScan output

bpatel2107 committed:
Merge pull request #448 from pyro-kinetics/feature/enforce_qn

Enforce quasineutrality when writing GK files

physycola committed:
Added writing of adiabatic electron flags in .set method

physycola committed:
[skip ci] Apply black changes

physycola committed:
Added warning in check_quasineutrality to allow for adiabatic electrons.

physycola committed:
[skip ci] Apply black changes

physycola committed:
Fixed fetching of rho_star from `.equilibrium` file

physycola committed:
Merge branch 'bugfix/cgyro_AE_conversion' of https://github.com/pyro-kinetics/pyrokinetics into bugfix/cgyro_AE_conversion

physycola committed:
Merge pull request #455 from pyro-kinetics/bugfix/cgyro_AE_conversion

Added writing of adiabatic electron flags in `.set` method, and updated `cgyro.py` to properly extract `rho_star` from the `.equilibrium` file.

physycola committed:
Changed the ordering of local_species when writing to GX input file for ease of handling of adiabatic electron species (GX does not like an adiabatic species being first).

physycola committed:
[skip ci] Apply black changes

physycola committed:
Empty commit to trigger remote tests

physycola committed:
Merge branch 'bugfix/gx_species_ordering' of https://github.com/pyro-kinetics/pyrokinetics into bugfix/gx_species_ordering

physycola committed:
Adding back in lines removed in previous commit.

physycola committed:
Give the user the ability to overwrite the base_directory from a pyroscan.json if loading the scan from a different location

physycola committed:
Removed erroneous f-string

bpatel2107 committed:
Fix bref units in get_gs2_geometry and unit dpsidr in CGYRO local_geometry

bpatel2107 committed:
[skip ci] Apply black changes

bpatel2107 committed:
Flake8 fix

bpatel2107 committed:
Merge remote-tracking branch 'origin/bugfix/gs2_geom_units' into bugfix/gs2_geom_units

bpatel2107 committed:
Pin scipy to <=1.15.3

bpatel2107 committed:
Use dpsidr from local_geometry instead of re-calculating it

bpatel2107 committed:
Fix FFprime test

ILexiconI committed:
Fixed comments at L1906 (#464)

Fixed and amended comments at L1906

bpatel2107 committed:
Fix error in dpsidr units in get_local_geometry

bpatel2107 committed:
Merge pull request #463 from pyro-kinetics/bugfix/gs2_geom_units

Bugfix/gs2 geom units

bpatel2107 committed:
Merge branch 'unstable' into bugfix/gx_species_ordering

bpatel2107 committed:
Test for name use sorted list

bpatel2107 committed:
Merge pull request #456 from pyro-kinetics/bugfix/gx_species_ordering

Ordering of local species when writing to GX input file

bpatel2107 committed:
Merge branch 'unstable' into bugfix/CGYRO_loading

bpatel2107 committed:
Merge pull request #457 from pyro-kinetics/bugfix/CGYRO_loading

Adding back in lines removed in previous commit.

bpatel2107 committed:
Merge branch 'unstable' into enhancement/pyroscan_working_directory

bpatel2107 committed:
Merge pull request #458 from pyro-kinetics/enhancement/pyroscan_working_directory

Enhancement/pyroscan working directory

bpatel2107 committed:
Check for required outputs only during verify and add warning for Spc/Poten data

bpatel2107 committed:
More sensible test on GKW coords

bpatel2107 committed:
Merge branch 'unstable' into enhancements/gkw_adiabatic_electrons

bpatel2107 committed:
Merge pull request #432 from pyro-kinetics/enhancements/gkw_adiabatic_electrons

GKW adiabatic electron plus more

dake0795 committed:
feat: add ELITEINP equilibrium reader with inline .eqin parser

dake0795 committed:
First attempt at ELITEINP reader for equilibrium

dake0795 committed:
Small changes to reader

dake0795 committed:
[skip ci] Apply black changes

dake0795 committed:
Add 2D interpolation for psi(R,Z) using R(psi,theta) and Z(psi,theta) from ELITEINP

dake0795 committed:
[skip ci] Apply black changes

dake0795 committed:
Addition of ELITEINP kinetics reader

dake0795 committed:
added impurities into kinetic reader

dake0795 committed:
[skip ci] Apply black changes

dake0795 committed:
small bug fixes

dake0795 committed:
[skip ci] Apply black changes

bpatel2107 committed:
Correct psiRZ data in eq and use rho from eq in kinetics

bpatel2107 committed:
[skip ci] Apply black changes

bpatel2107 committed:
Add test and template for ELITEINP

bpatel2107 committed:
Merge branch 'feature/eliteinp-io' of github.com:pyro-kinetics/pyrokinetics into feature/eliteinp-io

bpatel2107 committed:
Update docs

bpatel2107 committed:
Use small template and tweak verify and file opening

bpatel2107 committed:
[skip ci] Apply black changes

bpatel2107 committed:
Update error messaging for non HELENA inputs

bpatel2107 committed:
Remove pressure from ELITEINP check early on

bpatel2107 committed:
Flip if statement in eliteinp

d7919 committed:
Update metric.py

Fix typo in contravariant metric setting

bpatel2107 committed:
Merge pull request #467 from pyro-kinetics/d7919-patch-1

Update metric.py

bpatel2107 committed:
Merge pull request #465 from pyro-kinetics/feature/eliteinp-io

Feature/eliteinp io

d7919 committed:
Merge branch 'unstable' into feature/add_mxh_for_gs2

d7919 committed:
[skip ci] Apply black changes

d7919 committed:
Update gs2.py to remove c0_mxh and add n_mxh

d7919 committed:
Update gs2.py to fix typo

d7919 committed:
Update gs2.py to fix error message typo

bpatel2107 committed:
Take lowest kx in pyro output during pyroscan load

d7919 committed:
Update example_JETTO.py to allow GS2 input for non-Miller

Removes restriction on writing the GS2 input file when the geometry is not Miller

d7919 committed:
Update test_roundtrip.py to add GS2 to MXH roundtrip

d7919 committed:
Update gs2.py to allow LocalGeometryMXH

d7919 committed:
[skip ci] Apply black changes

d7919 committed:
Update gs2.py to use correct name for the number of moments

d7919 committed:
Update gs2.py to fix typo

d7919 committed:
Update gs2.py to define delta, s_kappa and s_delta in mxh

d7919 committed:
[skip ci] Apply black changes

d7919 committed:
Update gs2.py to add dZ0dr

d7919 committed:
Update gs2.py to convert mxh lists to arrays

d7919 committed:
Update gs2.py to fix name of instance referenced

d7919 committed:
Update gs2.py to be less general about how we convert list to array

d7919 committed:
[skip ci] Apply black changes

bpatel2107 committed:
Merge pull request #398 from pyro-kinetics/feature/add_mxh_for_gs2

Start to add MXH treatment for GS2

bpatel2107 committed:
Correct phase for moments in cgyro.py

physycola committed:
Removed setting of FIELD_PRINT_FLAG and MOMENT_PRINT_FLAG in cgyro.py

physycola committed:
Bugfix for when loading pyroscans without dimensions

bpatel2107 committed:
Merge pull request #471 from pyro-kinetics/bugfix/cgyro_moment_print_flag

Removed setting of FIELD_PRINT_FLAG and MOMENT_PRINT_FLAG in cgyro.py

bpatel2107 committed:
Merge pull request #470 from pyro-kinetics/bugfix/cgyro_moment_conjugate

Correct phase for moments in cgyro.py

bpatel2107 committed:
Merge pull request #468 from pyro-kinetics/bugfix/pyroscan_theta0

Bugfix for PyroScan loading with theta0

tomneiser committed:
Merge remote-tracking branch 'origin' into feature/tglf_satparams

tomneiser committed:
Merge remote-tracking branch 'origin' into feature/tglf_satparams

tomneiser committed:
refactor tglf satrule tests

tomneiser committed:
refactor tglf satrule tests

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
apply tglf satrules to pyroscan object

tomneiser committed:
apply tglf satrules to pyroscan object

tomneiser committed:
[skip ci] Apply black changes

tomneiser committed:
[skip ci] Apply black changes
